src/bt_peer_connection.cpp

```diff
@@ -348,6 +348,17 @@ namespace {
 		stats_counters().inc_stats_counter(counters::num_outgoing_dht_port);
 	}
 
+	template<class F, typename... Args>
+	void bt_peer_connection::extension_notify(F message, Args... args)
```

Comment:
I like this function. I imagine it could be made to replace some of the existing loops and calls to plugins too (in the future).